hack/install-go.sh

```diff
@@ -2,7 +2,8 @@
 
 destination=$1
 version=$(curl -s https://go.dev/dl/?mode=json | jq -r ".[0].version")
-tarball=$version.linux-amd64.tar.gz
+arch=$(uname -m | sed 's/x86_64/amd64/')
```

Comment:
I'm not sure I understand. 
If the machine running is `x86_64` then why would I want to substitute it to `amd64`?
Shouldn't it just be something like
```
arch=$(uname -m)
if [ "$arch" != "x86_64" ] && [ "$arch" != "amd64" ]; then
  echo "Unsupported architecture: $arch"
  exit 1
fi
```
?

Reply:
x86_64" and "amd64" are both valid terms used to refer to the same architecture. However, when you check the architecture via the command line using uname -m, it returns "x86_64". The Go programming language project, on the other hand, prefers the term "amd64." Therefore, if we are working with the amd64 architecture, we need to rename it accordingly.

Reply:
ah I see. Learned something new :) 
Please add this explanation to the commit desc.

Reply:
@RamLavi done. Thanks for the feedback.